src/lib.rs

```diff
@@ -87,21 +112,42 @@ fn parse_sealed(item: syn::Item) -> syn::Result<proc_macro2::TokenStream> {
 }
 
 // Care for https://gist.github.com/Koxiaet/8c05ebd4e0e9347eb05f265dfb7252e1#procedural-macros-support-renaming-the-crate
-fn parse_sealed_struct(strct: syn::ItemStruct) -> syn::Result<proc_macro2::TokenStream> {
-    let ident = &strct.ident;
+fn parse_sealed_trait(mut item_trait: syn::ItemTrait) -> syn::Result<proc_macro2::TokenStream> {
+    let trait_ident = &item_trait.ident;
+    let trait_generics = &item_trait.generics;
+    let seal = seal_name(trait_ident);
```

Comment:
You should also consider raw identifiers (`r#ident`). Use `.unraw()` for that. And cover this with a test case.

Reply:
Good catch!

Reply:
If you could submit a test case, I'd be very thankful

Reply:
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3b7d866ccd3e47f720072f0e6d8bcfce)

Usually, you won't meet such naming in a regualr handwritten code. Things change quite a lot when your library has a lot of code generation.